java/build/src/main/java/sleeper/build/status/ChunkStatus.java

```diff
@@ -16,61 +16,104 @@
 package sleeper.build.status;
 
 import java.io.PrintStream;
+import java.time.Instant;
+import java.util.Date;
 import java.util.Objects;
 
+import static sleeper.build.status.ValidationUtils.ignoreEmpty;
+
 public class ChunkStatus {
 
     private static final String COMPLETED = "completed";
     private static final String SUCCESS = "success";
+    private static final String IN_PROGRESS = "in_progress";
 
-    private final String chunk;
+    private final ProjectChunk chunk;
     private final String status;
     private final String conclusion;
+    private final Long runId;
+    private final String runUrl;
+    private final Instant runStarted;
+    private final String commitSha;
+    private final String commitMessage;
 
     private ChunkStatus(Builder builder) {
-        chunk = Objects.requireNonNull(ignoreEmpty(builder.chunk), "chunk must not be null");
+        chunk = Objects.requireNonNull(builder.chunk, "chunk must not be null");
         status = ignoreEmpty(builder.status);
         conclusion = ignoreEmpty(builder.conclusion);
+        runId = builder.runId;
+        runUrl = ignoreEmpty(builder.runUrl);
+        runStarted = builder.runStarted;
+        commitSha = ignoreEmpty(builder.commitSha);
+        commitMessage = ignoreEmpty(builder.commitMessage);
     }
 
     public static Builder builder() {
         return new Builder();
     }
 
-    public void report(PrintStream out) {
+    public void report(GitHubHead head, PrintStream out) {
+        out.println();
         if (conclusion != null) {
-            out.println(chunk + ": " + status + ", " + conclusion);
+            out.println(chunk.getName() + ": " + status + ", " + conclusion);
+        } else {
+            out.println(chunk.getName() + ": " + status);
+        }
+        if (runUrl != null) {
+            out.println("Run: " + runUrl);
+        }
+        if (runStarted != null) {
+            out.println("Started at: " + runStarted);
+        }
+        if (isRunForHead(head)) {
+            out.println("Build is for current commit");
         } else {
-            out.println(chunk + ": " + status);
+            if (commitMessage != null) {
+                out.println("Message: " + commitMessage);
+            }
+            if (commitSha != null) {
+                out.println("Commit: " + commitSha);
+            }
         }
     }
 
-    public boolean isFailCheck() {
-        return COMPLETED.equals(status) && !SUCCESS.equals(conclusion);
+    public boolean isFailCheckWithHead(GitHubHead head) {
+        return (COMPLETED.equals(status) && !SUCCESS.equals(conclusion))
+                || isWaitForOldBuildWithHead(head);
```

Comment:
Is it worth perhaps adding a check to see if the current running build is hanging? Maybe implement a configurable timeout that you use to measure against the started time of a build.

Reply:
Never mind, the MAX_RETRIES configuration solves this problem